sunpy/map/mapbase.py

```diff
@@ -2137,16 +2137,50 @@ def draw_rectangle(self, bottom_left, *, top_right=None, width: u.deg = None, he
         axes.add_patch(rect)
         return [rect]
 
-    @u.quantity_input
-    def draw_contours(self, levels: u.percent, axes=None, **contour_args):
+    def _process_levels_arg(self, levels, warn_units=False):
+        """
+        Accept a percentage or dimensionless or map unit input for contours.
+
+        ``warn_units`` allows legacy option of passing levels that aren't Quantities.
+        This can be removed in sunpy 3.1.
+        """
+        levels = np.atleast_1d(levels)
+        if not hasattr(levels, 'unit'):
+            if self.unit is None:
+                # No map units, so allow non-quantity through
+                return levels
+            elif warn_units:
+                warnings.warn('Passing contour levels without units is deprecated. '
```

Comment:
will percentage inputs be deprecated? if not this warning really should also say to either pass levels with units or as a percentage for when someone passes a list of numbers meaning to be percentages.

Reply:
No, they're not being deprecated (or removed), I'll updating the message.